src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -455,6 +430,44 @@ private Collection<FunctionDescriptor> ReadFunctions(ScriptHostConfiguration con
             return ReadFunctions(metadatas, descriptorProviders);
         }
 
+        /// <summary>
+        /// Determines which script should be considered the "primary" entry point script.
+        /// </summary>
+        internal static string DeterminePrimaryScriptFile(JObject functionConfig, string[] functionFiles)
+        {
+            if (functionFiles.Length == 1)
+            {
+                // if there is only a single file, that file is primary
+                return functionFiles[0];
```

Comment:
Note that functionFiles contains only the non function.json files, so this case handles the 90% case where there is just a function.json and a single file in the directory, say a DoIt.BAT. We don't want it to fail if there is only a single file.